### ISSUE TITLE
ZeroDivisionError in PlanetFocusInfo

### ISSUE THREAD

**Issue by thejze** [freeorion/freeorion#4652]
Bug Report
==========

Environment
-----------

Pre-packaged binary from OpenSuSE Build Service

rpm -q -i freeorion
Name        : freeorion
Version     : 0.5
Release     : lp154.1.1
Architecture: x86_64
Install Date: So 02 Jul 2023 20:20:23 CEST
Group       : Amusements/Games/Strategy/Turn Based
Size        : 34878210
License     : GPL-2.0-only
Signature   : DSA/SHA1, Mo 26 Jun 2023 00:07:01 CEST, Key ID 8f91b0e6c862b42c
Source RPM  : freeorion-0.5-lp154.1.1.src.rpm
Build Date  : Mo 26 Jun 2023 00:04:51 CEST
Build Host  : lamb04
Relocations : (not relocatable)
Packager    : https://www.suse.com/
Vendor      : obs://build.opensuse.org/games
URL         : https://freeorion.org/
Summary     : A turn-based space empire and galactic conquest (4X) computer game
Description :
FreeOrion is a turn-based space empire and galactic conquest (4X)
computer game. It is inspired by the tradition of the Master of Orion
games, but is not a clone or remake of that series or any other game.
Distribution: games / 15.4

* **FreeOrion Version**: 0.5
* **Operating System**: OpenSuSE 15.4
* **Graphic card used**: not relevant
* **Fetched as** 
  * Binary release
  

Description
-----------

During the gameplay the following Error appeared in the Message window. The gameplay continued, though.

````
[30 Jul 12:44:22] AI_4: AI_Error: AI script error : FreeOrionAI.py:generateOrders():345  - Exception float division by zero while trying to generate_resources_orders
Traceback (most recent call last):
  File "/usr/share/freeorion/default/python/AI/FreeOrionAI.py", line 342, in generateOrders
    action()
  File "/usr/share/freeorion/default/python/AI/ResourcesAI.py", line 751, in generate_resources_orders
    focus_manager.set_planetary_foci(reporter)
  File "/usr/share/freeorion/default/python/AI/ResourcesAI.py", line 149, in set_planetary_foci
    self.set_influence_focus()
  File "/usr/share/freeorion/default/python/AI/ResourcesAI.py", line 434, in set_influence_focus
    [
  File "/usr/share/freeorion/default/python/AI/ResourcesAI.py", line 435, in <listcomp>
    (pinfo.evaluate(INFLUENCE), pinfo)
  File "/usr/share/freeorion/default/python/AI/ResourcesAI.py", line 121, in evaluate
    return self.rating_below_best(focus) / sqrt_population
ZeroDivisionError: float division by zero
Local context: {'self': PlanetFocusInfo(P_11619<Orthe γ II>), 'focus': 'FOCUS_INFLUENCE', 'sqrt_population': 0.0}
````
Expected Result
---------------
No error message. 
[error230730.log](https://github.com/freeorion/freeorion/files/12209018/error230730.log)
[game 230730.zip](https://github.com/freeorion/freeorion/files/12209019/game.230730.zip)
[freeorion.log](https://github.com/freeorion/freeorion/files/12209021/freeorion.log)

Steps to reproduce
------------------
Can't say.

Best regards
Thomas

**Comment by geoffthemedio** [comment]
Similar to #4570

**Comment by geoffthemedio** [comment]
@Cjkjvfnby ?

**Comment by Cjkjvfnby** [comment]
Looks like the same issue.  The fix is  already merged into 0.5.1

**Comment by Vezzra** [comment]
@Cjkjvfnby, has the fix been backported to the 0.5 release branch?

**Comment by Cjkjvfnby** [comment]
@Vezzra 
 https://github.com/freeorion/freeorion/pull/4596
